src/Host/UserProfile/Impl/ServiceLoggerProvider.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.R.Host.UserProfile {
+    class ServiceLoggerProvider : ILoggerProvider {
+
+        private readonly StreamWriter _writer;
+        private readonly List<ServiceLogger> _loggers = new List<ServiceLogger>();
+
+        public ServiceLoggerProvider()
+                : this(File.CreateText(GetLogFileName())) {
+        }
+
+        public ServiceLoggerProvider(StreamWriter writer) {
+            _writer = writer;
+        }
+
+        private static string GetLogFileName() {
+            return Path.Combine(Path.GetTempPath(), $@"Microsoft.R.Host.UserProfile_{DateTime.Now:yyyyMdd_HHmmss}_pid{Process.GetCurrentProcess().Id}.log");
+        }
+
+        public ILogger CreateLogger(string categoryName) {
```

Comment:
#2477

---
In reply to: [82319599](https://github.com/Microsoft/RTVS/pull/2432#discussion_r82319599) [](ancestors = 82319599)